thrust/detail/config/host_device.h

```diff
@@ -42,3 +42,9 @@
 
 #endif
 
+/**
+ * \def THRUST_HOST_DEVICE
+ * This macro is used replace __host__ __device__, with a clang-format friendly
+ * replacement
+ */
+#define THRUST_HOST_DEVICE __host__ __device__
```

Comment:
From my point of view it matters a lot, as it greatly enlarges function definitions. 

If we have the convention that all combinations of {`__host__`, `__device__`} are on the same line before the function declaration, it greatly reduces mental overhead

Actually I am really in favor of pushing all those definitions into libcu++ and have a single source of truth

Reply:
My 2c, I prefer having the attributes / annotations on a separate line from the function definition. It makes it much easier to parse the function def at a glance, since the return type is always as the beginning of the line. Having to parse a variable number of tokens from the start of a function def adds extra steps when skimming code.